src/python/foglamp/translators/omf_translator.py

```diff
@@ -944,40 +936,36 @@ def omf_object_creation():
 
                         position_update(new_position)
 
+                        await  _update_statistics()
+
     except Exception as e:
         message = _message_list["e000004"].format(e)
 
-        _log.error(message)
+        _logger.exception(message)
         raise Exception(message)
 
 async def _update_statistics():
-    global _num_readings
-    global _num_discarded_readings
+
     await statistics.update_statistics_value('SENT', _num_sent)
-    _num_readings = 0
-    await statistics.update_statistics_value('UNSENT', _num_unsent)
```

Comment:
You are setting the _num_unsent in send_info_to_omf method, then should this also not go in statistics table?